### PR TITLE
Vip go standards

### PR DESCRIPTION
**Pull request by a6software**
Fixes all warnings, and most errors. 

The errors that remain are:

```
➜  GDPR git:(vip-go-standards) ./bin/phpcs -n --severity=1 --standard=./phpcs.xml.dist .
...E.....................E....................... 49 / 49 (100%)



FILE: /Users/christophermoss/Development/boxuk/GDPR/includes/helper-functions.php
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
FOUND 5 ERRORS AFFECTING 4 LINES
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
  93 | ERROR | Due to server-side caching, server-side based client related logic might not work. We recommend implementing client side logic in JavaScript instead.
     |       | (WordPress.VIP.RestrictedVariables.cache_constraints___COOKIE)
  94 | ERROR | Due to server-side caching, server-side based client related logic might not work. We recommend implementing client side logic in JavaScript instead.
     |       | (WordPress.VIP.RestrictedVariables.cache_constraints___COOKIE)
 132 | ERROR | Due to server-side caching, server-side based client related logic might not work. We recommend implementing client side logic in JavaScript instead.
     |       | (WordPress.VIP.RestrictedVariables.cache_constraints___COOKIE)
 132 | ERROR | Due to server-side caching, server-side based client related logic might not work. We recommend implementing client side logic in JavaScript instead.
     |       | (WordPress.VIP.RestrictedVariables.cache_constraints___COOKIE)
 133 | ERROR | Due to server-side caching, server-side based client related logic might not work. We recommend implementing client side logic in JavaScript instead.
     |       | (WordPress.VIP.RestrictedVariables.cache_constraints___COOKIE)
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------


FILE: /Users/christophermoss/Development/boxuk/GDPR/public/class-gdpr-public.php
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
FOUND 9 ERRORS AFFECTING 7 LINES
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 179 | ERROR | Due to server-side caching, server-side based client related logic might not work. We recommend implementing client side logic in JavaScript instead.
     |       | (WordPress.VIP.RestrictedVariables.cache_constraints___COOKIE)
 179 | ERROR | Due to server-side caching, server-side based client related logic might not work. We recommend implementing client side logic in JavaScript instead.
     |       | (WordPress.VIP.RestrictedVariables.cache_constraints___COOKIE)
 180 | ERROR | Due to server-side caching, server-side based client related logic might not work. We recommend implementing client side logic in JavaScript instead.
     |       | (WordPress.VIP.RestrictedVariables.cache_constraints___COOKIE)
 180 | ERROR | Due to server-side caching, server-side based client related logic might not work. We recommend implementing client side logic in JavaScript instead.
     |       | (WordPress.VIP.RestrictedVariables.cache_constraints___COOKIE)
 249 | ERROR | Due to server-side caching, server-side based client related logic might not work. We recommend implementing client side logic in JavaScript instead.
     |       | (WordPress.VIP.RestrictedVariables.cache_constraints___COOKIE)
 252 | ERROR | Due to server-side caching, server-side based client related logic might not work. We recommend implementing client side logic in JavaScript instead.
     |       | (WordPress.VIP.RestrictedVariables.cache_constraints___COOKIE)
 339 | ERROR | Due to server-side caching, server-side based client related logic might not work. We recommend implementing client side logic in JavaScript instead.
     |       | (WordPress.VIP.RestrictedVariables.cache_constraints___COOKIE)
 349 | ERROR | Due to server-side caching, server-side based client related logic might not work. We recommend implementing client side logic in JavaScript instead.
     |       | (WordPress.VIP.RestrictedVariables.cache_constraints___COOKIE)
 360 | ERROR | Due to server-side caching, server-side based client related logic might not work. We recommend implementing client side logic in JavaScript instead.
     |       | (WordPress.VIP.RestrictedVariables.cache_constraints___COOKIE)
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------

Time: 4.29 secs; Memory: 22MB
```

All of these relate to checking the status of existing cookies:

* `isset`
* `empty`

Access, e.g.:

https://github.com/boxuk/GDPR/blob/develop/public/class-gdpr-public.php#L356

Or removing cookies:

https://github.com/boxuk/GDPR/blob/develop/public/class-gdpr-public.php#L259

There are ways to do this using JS, and the plugin itself does come with some helpful JS implementations of PHP functions we use, such as:

https://github.com/boxuk/GDPR/blob/develop/src/js/public/gdpr-public.js#L18

However, there is one major pain point for which I cannot figure out a solution.

In order to remove the tracking script for users who have not consented to the cookie policy, we are currently using a WP (jetpack specifically) hook to enable do not track:

https://github.com/boxuk/peake-wp-content/blob/develop/client-mu-plugins/gdpr/class-gdpr-cookie-tracking.php#L27

I cannot figure out a way to do this solely using JS. In order for JS to run, the DOM needs to be loaded, and this hook takes place before that happens. Save for using PHP V8  (which I have no experience of) I'm not sure this is possible.

Fixing these issues as suggested by VIP Bot (re-writing using client side JS) is a substantial piece of work to the best of my knowledge.

### CODE REVIEW

**Comment by a6software** [comment]
Note - build fails for reasons described above.

File: includes/class-gdpr-requests.php [hunk 1]
```diff
@@ -164,7 +164,7 @@ protected function confirm_request( $key ) {
 
 		$user = get_user_by( 'email', $email );
 
-		// phpcs:disable WordPress.DB.SlowDBQuery.slow_db_query_meta_key
+		// phpcs:disable WordPress.VIP.SlowDBQuery.slow_db_query_meta_key
```
Comment (jenkoian) [inline includes/class-gdpr-requests.php]: Worth a note explaining why, like you've done elsewhere?

File: includes/class-gdpr-requests.php [hunk 2]
```diff
@@ -265,7 +265,7 @@ protected function add_to_requests( $email, $type, $data = null, $confirmed = fa
 		/**
 		 * Remove user from the requests if it did not confirm in 2 days.
 		 */
-		// phpcs:disable WordPress.DB.SlowDBQuery.slow_db_query_meta_key
+		// phpcs:disable WordPress.VIP.SlowDBQuery.slow_db_query_meta_key
```
Comment (jenkoian) [inline includes/class-gdpr-requests.php]: As above, worth a note?

File: public/class-gdpr-requests-public.php [hunk 3]
```diff
@@ -59,9 +59,11 @@ public function delete_user( $user, $index ) {
 	 * @return mixed        Print the form html.
 	 */
 	public static function request_form( $type, $submit_button_text = '' ) {
+		// phpcs:disable WordPressVIPMinimum.Variables.VariableAnalysis.UndefinedVariable
```
Comment (jenkoian) [inline public/class-gdpr-requests-public.php:63]: Worth a note to explain why, as you've done elsewhere?

File: public/class-gdpr-requests-public.php [hunk 4]
```diff
@@ -84,7 +86,9 @@ public function send_request_email() {
 			);
 		}
 
+		// phpcs:disable WordPressVIPMinimum.Variables.VariableAnalysis.UndefinedVariable
```
Comment (jenkoian) [inline public/class-gdpr-requests-public.php:95]: Worth a note to explain why, as you've done elsewhere?

File: includes/class-gdpr-requests.php [hunk 5]
```diff
@@ -164,7 +164,8 @@ protected function confirm_request( $key ) {
 
 		$user = get_user_by( 'email', $email );
 
-		// phpcs:disable WordPress.VIP.SlowDBQuery.slow_db_query_meta_key
+		// Added VIP equivalent sniff, as per original plugin PHPCS disabled sniff.
```
Comment (jenkoian) [inline includes/class-gdpr-requests.php:167]: Oh the plugin was already doing this? I was wondering what it was that was causing the warning?

Comment (a6software) [reply includes/class-gdpr-requests.php:167]: Yeah, the original plugin is(, or I think was) following a set standard, which is not identical to ours, but still somehow conforms to VIP afaik. 

Not sure, this plugin is weird in many and various ways.

**Comment by a6software** [comment]
Not had much luck replacing these bits and pieces with JS. It's like a ball of yarn. Every time I pull one bit, a larger and larger section needs replacing. Still on it, just very frustrating.

**Review by jenkoian — APPROVED** [review]

**Comment by a6software** [comment]
To clarify, the cookie errors were acceptable in the eyes of VIP support. They have now been suppressed.

**Pull request merged**